benches/regalloc.rs

```diff
@@ -0,0 +1,56 @@
+//! Criterion-based benchmark target that computes insts/second for
+//! arbitrary inputs.
```

Comment:
I wouldn't exactly call this even close to a realistic benchmark as normal inputs have far more structure.

Reply:
Well, the SSA generator does build interesting CFGs (it will produce nested loops, control-flow diamonds and the like) and I tried to get the balance of local vs long-distance uses to be reasonable, but I agree that randomly-pick-a-vreg-for-a-use is not going to be as good as a real function body. The question is whether throughput depends on specific properties of real code or whether this input with similar bulk properties (interference, live-range length, CFG complexity) can give us a good first estimate.

The intent, FWIW, was to get an early indication of whether I was even on the right path before spending more time integrating into Cranelift. I'm about halfway into the Cranelift-specific part now, so hopefully we'll have better numbers soon :-) I certainly expect we'll carefully evaluate Cranelift with real benchmarks (e.g. Sightglass) before switching over.